src/SixLabors.Fonts/GlyphPositioningCollection.cs

```diff
@@ -17,16 +18,18 @@ internal sealed class GlyphPositioningCollection : IGlyphShapingCollection
     /// <summary>
     /// Contains a map the index of a map within the collection, non-sequential codepoint offsets, and their glyph ids, point size, and mtrics.
     /// </summary>
-    private readonly List<GlyphPositioningData> glyphs = new();
+    private GlyphPositioningData[] glyphs = ArrayPool<GlyphPositioningData>.Shared.Rent(16);
 
     /// <summary>
     /// Initializes a new instance of the <see cref="GlyphPositioningCollection"/> class.
     /// </summary>
     /// <param name="textOptions">The text options.</param>
     public GlyphPositioningCollection(TextOptions textOptions) => this.TextOptions = textOptions;
 
+    ~GlyphPositioningCollection() => ArrayPool<GlyphPositioningData>.Shared.Return(this.glyphs);
```

Comment:
In C#, relying on a destructor (finalizer) without implementing `IDisposable` is typically discouraged and rarely sufficient. 

- The finalizer (`~ClassName`) runs during garbage collection.
- The timing of garbage collection is non-deterministic, so you cannot guarantee when (or even if) the finalizer will execute.
- Adding a finalizer to a class can degrade performance because the object is placed in a finalization queue, requiring at least two garbage collection cycles to be reclaimed.